skiplist/src/list.rs

```diff
@@ -210,20 +212,68 @@ impl<C: KeyComparator> Skiplist<C> {
 
         let height = self.random_height();
         let node_offset = Node::alloc(&self.inner.arena, key, value, height);
-        while height > list_height {
-            match self.inner.height.compare_exchange_weak(
-                list_height,
-                height,
-                Ordering::SeqCst,
-                Ordering::SeqCst,
-            ) {
-                Ok(_) => break,
-                Err(h) => list_height = h,
+        if self.allow_concurrent_write {
+            while height > list_height {
+                match self.inner.height.compare_exchange_weak(
+                    list_height,
+                    height,
+                    Ordering::SeqCst,
+                    Ordering::SeqCst,
+                ) {
+                    Ok(_) => break,
+                    Err(h) => list_height = h,
+                }
+            }
+        } else {
+            // There is no need to use CAS for single thread writing.
```

Comment:
we plan to support the multiple thread writing, it's necessary now? We may need to delete in future.

Reply:
If treat AgateDB as a fundamental kv engine on cloud, maybe it need both single thread write and multi-threads write.